### ISSUE TITLE
[knx] Control item doesn’t respond to read request from the bus - throwing an exception

### ISSUE THREAD

**Issue by Joerg-Dr** [openhab/openhab-addons#12446]
### Prerequisites
A configured KNX thing with a channel of type *-control (e.g. switch-control)
A link of the channel to an item in OH3 (e.g. Switch)
The state if the item is defined (e.g. ON)

### Expected Behavior
If the bus sends a read request to the main GA of the control channel, OH responds with a read response telling the current state of the item.

### Current Behavior
If the bus sends a read request to the main GA of the control item, there is no response from OH3.
Internally there is an exception in the KNX binding when handling the GroupValueRead request.

**This bug might be quite new, in former versions of the binding I did not observe this problem.**

### Use Case
There are several situations where read requests on the bus (GroupValueRead) are necessary (e.g. to read the current state of a device).

### Possible Solution
None.

### Steps to Reproduce (for Bugs)
Configure a device to read from the main GA in ETS.
Configure OH and the knx binding as described in the prerequisites.
Start the group monitor in ETS.
Select the main GA of the control item.
Read from the address -> A read request is seen on the bus.
Wait some seconds.
Verify that there is no response to be seen on the bus.

Activating the KNX binding log in 'log4j2.xml' shows an exception in the KNX binding:
<Logger level="TRACE" name="org.openhab.binding.knx"/>

```
Read request on GA 5/5/3 (DPT 1.001)  (switch control)
-----------------------------------------------------

18:03:14.180 [TRACE] [knx.internal.client.AbstractKNXClient] - Received a Group Read Request telegram from '1.1.251' to '5/5/3' with value '[]'
18:03:14.185 [TRACE] [x.internal.handler.DeviceThingHandler] - onGroupRead Thing 'knx:device:4b5cd903:3db268168c' received a GroupValueRead telegram from '1.1.251' for destination '5/5/3'
18:03:14.191 [TRACE] [x.internal.handler.DeviceThingHandler] - onGroupRead isControl -> postCommand
18:03:14.195 [TRACE] [x.internal.handler.DeviceThingHandler] - onGroupRead size '1'
18:03:14.199 [TRACE] [x.internal.handler.DeviceThingHandler] - onGroupRead respondToKNX '5/5/3'
18:03:14.203 [TRACE] [knx.internal.client.AbstractKNXClient] - respondToKNX groupAddress '5/5/3', responseSpec 'org.openhab.binding.knx.internal.channel.WriteSpecImpl@d585f4'
18:03:14.207 [TRACE] [knx.internal.client.AbstractKNXClient] - sendToKNX mappedValue: 'on' groupAddress: '5/5/3'
18:03:14.211 [WARN ] [ommon.WrappedScheduledExecutorService] - Scheduled runnable ended with an exception:
java.lang.NullPointerException: null
        at tuwien.auto.calimero.device.ProcessCommunicationResponder.send(ProcessCommunicationResponder.java:256) ~[?:?]
        at tuwien.auto.calimero.device.ProcessCommunicationResponder.write(ProcessCommunicationResponder.java:249) ~[?:?]
        at tuwien.auto.calimero.device.ProcessCommunicationResponder.write(ProcessCommunicationResponder.java:226) ~[?:?]
        at org.openhab.binding.knx.internal.client.AbstractKNXClient.sendToKNX(AbstractKNXClient.java:460) ~[?:?]
        at org.openhab.binding.knx.internal.client.AbstractKNXClient.respondToKNX(AbstractKNXClient.java:439) ~[?:?]
        at org.openhab.binding.knx.internal.handler.DeviceThingHandler.lambda$8(DeviceThingHandler.java:297) ~[?:?]
        at org.openhab.binding.knx.internal.handler.DeviceThingHandler.withKNXType(DeviceThingHandler.java:148) ~[?:?]
        at org.openhab.binding.knx.internal.handler.DeviceThingHandler.sendGroupValueResponse(DeviceThingHandler.java:286) ~[?:?]
        at org.openhab.binding.knx.internal.handler.DeviceThingHandler.lambda$10(DeviceThingHandler.java:318) ~[?:?]
        at org.openhab.binding.knx.internal.handler.DeviceThingHandler.withKNXType(DeviceThingHandler.java:148) ~[?:?]
        at org.openhab.binding.knx.internal.handler.DeviceThingHandler.onGroupRead(DeviceThingHandler.java:312) ~[?:?]
        at org.openhab.binding.knx.internal.client.AbstractKNXClient$1.lambda$1(AbstractKNXClient.java:115) ~[?:?]
        at org.openhab.binding.knx.internal.client.AbstractKNXClient.lambda$8(AbstractKNXClient.java:258) ~[?:?]
        at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515) ~[?:?]
        at java.util.concurrent.FutureTask.run(FutureTask.java:264) ~[?:?]
        at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:304) ~[?:?]
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128) [?:?]
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628) [?:?]
        at java.lang.Thread.run(Thread.java:834) [?:?]

```
### Context
I have a Domovea control and visualisation device, where the system issues read requests frequently to update the visualisation.

### My Environment
I'm using openHAB 3.2.0 running from a Raspberry Pi 4.
For KNX network connectivity I'm using a KNX IP Router 750 by Weinzierl Engineering.
For programming and testing the KNX installation I'm using the ETS5 professional version.

**Comment by Joerg-Dr** [comment]
Can anyone give me some help with this problem?
At the moment I work with the release build 3.2.0.
Have there been changes to the KNX binding in the latest OpenHAB build 3.3.0?

**Comment by schinken** [comment]
I have the same Issue, I think.

I have a number-control on **0/2/7**:
![image](https://user-images.githubusercontent.com/187176/160107450-d9ad7d0e-cb5e-4e3f-ae3e-6d1f95479334.png)

Sending values on change by rule does work:

![image](https://user-images.githubusercontent.com/187176/160107550-5578657d-0338-4db1-b167-bddde10871f3.png)
![image](https://user-images.githubusercontent.com/187176/160107578-181a02b3-ba83-4820-9b21-e215de0a3b22.png)

But when I request a Value in Group Monitor (**GroupValueRead**)
![image](https://user-images.githubusercontent.com/187176/160107709-74b7f6f2-53ee-405c-bf6d-082538c1f924.png)

I also get an exception in openhab.log:
![image](https://user-images.githubusercontent.com/187176/160107745-3d20eee4-c1f5-4375-8746-5bf9a92bafc1.png)

I'm currently running openHAB 3.2.0. And as I'm writing this issue, i think i found a fix in OH3.3.0: https://github.com/openhab/openhab-addons/pull/12074

**Comment by Joerg-Dr** [comment]
Thank you, good to know that it has been fixed in OH 3.3.0.
As I have seen, the reason was a new parameter, that had been invented in the latest version of Calimero, but had not been initialized by the binding.

There is a test build available, so I will test it and when everything is ok, I will close this issue.
[https://openhab.jfrog.io/ui/native/libs-pullrequest-local/org/openhab/addons/bundles/org.openhab.binding.knx/3.3.0-SNAPSHOT/org.openhab.binding.knx-3.3.0-SNAPSHOT.jar](url)

**Comment by Joerg-Dr** [comment]
Ok, it seems to work now. GroupReadRequests are answered as expected.

For installation you have to uninstall the KNX binding first, put the .jar file in the addons folder and reboot the machine (or restart OpenHAB, whatever is convenient for you).

After restart, OpenHAB might throw the following error:

> Unresolved requirement: Import-Package: gnu.io

You can resolve this by manually install the serial binding in the console:

> feature:install openhab-transport-serial

After a while the KNX gateway should work now like expected.
